src/bootstrap/src/core/build_steps/install.rs

```diff
@@ -23,7 +23,12 @@ const SHELL: &str = "sh";
 /// characters and on `C:\` paths, so normalize both of them away.
 fn sanitize_sh(path: &Path) -> String {
     let path = path.to_str().unwrap().replace('\\', "/");
-    return change_drive(unc_to_lfs(&path)).unwrap_or(path);
+    return if is_cygwin() { path } else { change_drive(unc_to_lfs(&path)).unwrap_or(path) };
+
+    fn is_cygwin() -> bool {
+        // ref. https://cygwin.com/pipermail/cygwin/2022-February/250802.html
+        env::var("OSTYPE").is_ok_and(|v| v.to_lowercase().contains("cygwin"))
+    }
```

Comment:
Initially, I wanted to handle it within `TargetSelection` implementation alongside other platform detection functions. But then I decided not to do it to avoid requiring a `TargetSelection` instance to run this function in `sanitize_sh`. I revisited it and put in some extra effort to improve it. Let me know if that works for you.

Reply:
That looks good to me, thanks!